base/exports.jl

```diff
@@ -679,7 +679,7 @@ export
 # errors
     backtrace,
     catch_backtrace,
-    catch_stack,
```

Comment:
Probably we want a deprecation/alias for `catch_stack`?

Reply:
`catch_stack` has only existed for a few months during the current dev cycle, so I don't have any qualms about removing it entirely. In fact that's one of the more important points of this PR: to clean it up so that the API is in a better state for 1.1.